src/controls/controls.actions.ts

```diff
@@ -351,51 +351,43 @@ import { renderCircleControl, renderSquareControl } from './controls.render';
    * @private
    */
   function skewObjectY(eventData, transform, x, y) {
-    var target = transform.target,
+    const target = transform.target,
       // find how big the object would be, if there was no skewX. takes in account scaling
       dimNoSkew = target._getTransformedDimensions({
         skewX: target.skewX,
-        skewY: 0,
+        skewY: transform.skewY,
       }),
-      localPoint = getLocalPoint(
+      diff = getLocalPoint(
         transform,
-        transform.originX,
-        transform.originY,
+        target.originX,
+        target.originY,
         x,
         y
-      ),
+      ).subtract(new Point(transform.offsetX, transform.offsetY)),
       // the mouse is in the center of the object, and we want it to stay there.
       // so the object will grow twice as much as the mouse.
       // this makes the skew growth to localPoint * 2 - dimNoSkew.
-      totalSkewSize = Math.abs(localPoint.y * 2) - dimNoSkew.y,
-      currentSkew = target.skewY,
-      newSkew;
-    if (totalSkewSize < 2) {
-      // let's make it easy to go back to position 0.
-      newSkew = 0;
-    } else {
-      newSkew = radiansToDegrees(
+      totalSkewSize = diff.y * 2,
+      currentSkew = target.skewY;
+    const skewSign =
+      (transform.originX === LEFT && transform.originY === BOTTOM) ||
+      (transform.originX === RIGHT && transform.originY === TOP) ||
+      targetHasOneFlip(target)
+        ? -1
+        : 1;
```

Comment:
not sure if this is intentional.
SkewSign is supposed to change sign every time one of those conditions are true.
So this is not equivalent at all, i know is a WIP, but in case you missed that this is a completely change of direction in behavior, i ll tell you or you could waste ours on it

Reply:
thanks
didn't really understand this logic yet